### PR TITLE
refactor: improve app update experience

### LINKED ISSUE #1448: Arrow symbol in tray
### 🔍 Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### 📝 Description

Gitify started to show an arrow in the tray. I could not find what this means and I also don't like applications taking up a lot of space in the tray.

### 🪜 Steps To Reproduce

Open Gitify 5.12.0

### Gitify Version

Other

### Operating System

macOS

### GitHub Account

GitHub Cloud

### 📸 Screenshots

image

### PR DESCRIPTION
**Pull request by setchy**
Iterate on the enhancements added in https://github.com/gitify-app/gitify/pull/1397

- Introduce https://github.com/electron/update-electron-app to handle the update lifecycle (interval, user confirmation dialog, release notes)
- Fix issues with context menu lifecycle state for ad-hoc updates
- Add a set of new tray icons for visual update queues (resolves #1448)

### CODE REVIEW

**Comment by setchy** [comment]
Added new tray icons to show a visual indicator for "update available", as well as updating the tray tooltip

| Idle | Active |
| - | - |
| tray-idle-update@2x | tray-active-update@2x |

File: package.json [hunk 1]
```diff
@@ -107,7 +107,7 @@
     "clsx": "2.1.1",
     "date-fns": "3.6.0",
     "electron-log": "5.1.7",
-    "electron-updater": "6.3.2",
+    "electron-updater": "6.2.1",
```
Comment (bmulholland) [inline package.json:110]: Intentional downgrade?

**Review by bmulholland — APPROVED** [review]
Nice to use a lib for it, presumably they've thought it all through more than we have

File: package.json (hunk 1 above)
Comment (setchy) [reply package.json:110]: Actually no, good catch. Will correct that

**Comment by afonsojramos** [comment]
Definitely way prettier 🚀

File: package.json [hunk 2]
```diff
@@ -117,7 +117,8 @@
     "react-router-dom": "6.26.0",
     "tailwind-merge": "2.4.0",
     "ts-loader": "9.5.1",
-    "typescript": "5.5.4"
+    "typescript": "5.5.4",
+    "update-electron-app": "^3.0.0"
```
Comment (afonsojramos) [inline package.json:121]:
```suggestion
    "update-electron-app": "3.0.0"
```
We should be pinning this right?

File: src/components/settings/SettingsFooter.tsx [hunk 3]
```diff
@@ -42,23 +31,6 @@ export const SettingsFooter: FC<ISettingsFooter> = ({
...
-              </span>
-            ) : (
-              <span title="You are using the latest version">
-                <CheckCircleFillIcon
-                  size={Size.XSMALL}
-                  className={IconColor.GREEN}
-                />
-              </span>
-            )}
-          </span>
```
Comment (afonsojramos) [inline src/components/settings/SettingsFooter.tsx:61]: Thanks 🙏 Didn't like it either 👀

**Review by afonsojramos — APPROVED** [review]

File: src/components/settings/SettingsFooter.tsx (hunk 3 above)
Comment (setchy) [reply src/components/settings/SettingsFooter.tsx:61]: Yeah.... That didn't turn out as I'd hoped 😅

**Pull request merged**